Geometry/TrackerCommonData/plugins/DDTrackerIrregularRingAlgo.cc

```diff
@@ -0,0 +1,249 @@
+#include "DetectorDescription/Core/interface/DDAlgorithm.h"
+#include "DetectorDescription/Core/interface/DDAlgorithmFactory.h"
+#include "DataFormats/Math/interface/angle_units.h"
+
+#include <string>
+#include <vector>
+
+/*
+  Tilts and positions n copies of a module at prescribed phi values
+  within a ring. The module can also be flipped if requested.
+
+  (radius, Phi, Z) refers to the cylindrical coordinates in the global frame of reference.
+  
+  A module's tilt angle is defined with respect to the global frame of reference's Z axis.
+  Example, in the outer tracker : For a straight barrel module, tiltAngle = 0°.
+  For a module in the endcaps, tiltAngle = 90°.
+  tiltAngle ∈ [0, 90°].
+  Please note that parameter tiltAngle has to be set regardless of any sign consideration,
+  to the absolute value of the module's tilt angle.
+
+  == Example of use : ==
+
+  <Algorithm name="track:DDTrackerIrregularRingAlgo">
+  <rParent name="tracker:Ring5Layer1Plus"/>
+  <String name="ChildName" value="tracker:BModule5Layer1"/>
+  <Numeric name="N" value="9"/>
+  <Numeric name="StartCopyNo" value="1"/>
+  <Numeric name="IncrCopyNo" value="2"/>
+  <Numeric name="RangeAngle" value="360*deg"/>
+  <Numeric name="StartAngle" value="90*deg"/>
+  <Numeric name="Radius" value="247"/>
+  <Vector name="Center" type="numeric" nEntries="3">0,0,-5.45415</Vector>
+  <Numeric name="IsZPlus" value="1"/>
+  <Numeric name="TiltAngle" value="47*deg"/>
+  <Numeric name="IsFlipped" value="1"/>
+  </Algorithm>
+*/
+
+using namespace std;
+using namespace angle_units::operators;
+
+class DDTrackerIrregularRingAlgo : public DDAlgorithm {
+public:
+  // Constructor and Destructor
+  DDTrackerIrregularRingAlgo();
+  ~DDTrackerIrregularRingAlgo() override;
+
+  void initialize(const DDNumericArguments& nArgs,
+                  const DDVectorArguments& vArgs,
+                  const DDMapArguments& mArgs,
+                  const DDStringArguments& sArgs,
+                  const DDStringVectorArguments& vsArgs) override;
+
+  void execute(DDCompactView& cpv) override;
+
+private:
+  int n;                  //Number of copies
+  int startCopyNo;        //Start Copy number
+  int incrCopyNo;         //Increment in Copy number
+  double rangeAngle;      //Range in Phi angle
+  double startAngle;      //Start Phi angle
+  double radius;          //Radius
+  vector<double> center;  //Phi values
+  vector<double> phiAngles;
+  vector<double> radiusValues;
+  vector<double> yawAngles;
+  bool isZPlus;      //Is Z positive ?
+  double tiltAngle;  //Module's tilt angle (absolute value)
+  bool isFlipped;    //Is the module flipped ?
+
+  string idNameSpace;  //Namespace of this and ALL sub-parts
+  string childName;    //Child name
+};
+
+DDTrackerIrregularRingAlgo::DDTrackerIrregularRingAlgo() {
+  LogDebug("TrackerGeom") << "DDTrackerIrregularRingAlgo info: Creating an instance";
+}
+
+DDTrackerIrregularRingAlgo::~DDTrackerIrregularRingAlgo() {}
```

Comment:
could be `default`.

Reply:
@mmusich Do you mean putting this:`DDTrackerIrregularRingAlgo::~DDTrackerIrregularRingAlgo()  = default;` or something else?

Reply:
yes, this destructor is doing nothing